src/spawn.coffee

```diff
@@ -30,12 +30,17 @@ procNotFoundError = (error, cmd) ->
   error.message = "Unable to find #{cmd}"
   error
 
-interpolatePort = (port) ->
-  (arg) ->
-    arg.replace '%port%', port
+processPorts = {}
+interpolatePorts = (args, processName, port) ->
+  processPorts["#{processName}.port"] = port
```

Comment:
Agreed. I was considering some way of throwing an error if the key didn't exist.